### DISCUSSION TITLE
Installation help

### DISCUSSION THREAD

**Discussion by southbayjay** [toutjavascript/FoooXus-Fooocus-Extender#10] [General]
I hate to say this but the instructions on how to install this are very vague. I'm not a coder but consider myself "knowledgeable" enough to install apps like these but I'm completely lost trying to install this. It looks fantastic and useful but I just can't figure out the install. There is talk about using the .exe file to install it but for the life of me I can't seem to find it. 

Please help

**Comment by simartem** [comment]
instructions as follows (official) works perfectly. Most important point is to do in same order and not to miss any parts.

Go create some directory like "C:\test" or anything.

For example : 

C:\test> git clone https://github.com/toutjavascript/FoooXus-Fooocus-Extender.git
C:\test> cd FoooXus-Fooocus-Extender

# Activate the virtual python environment
-> C:\test\FoooXus-Fooocus-Extender> python -m venv venv
--> C:\test\FoooXus-Fooocus-Extender> venv\Scripts\activate

# Install requirements
--> C:\test\FoooXus-Fooocus-Extender> pip install -r requirements.txt

# First run of the FoooXus app to initialize
--> C:\test\FoooXus-Fooocus-Extender> python foooxus.py

After applying the above commands, DO MODIFY "config.json" for your installed Focus directory and when defining the path, use double backslash. Check your path is correct and you have used DOUBLE BACKSLASH.

Example for my config.json 

`    "fooocus-directory": "C:\\Users\\user2\\Documents\\Focus",`

SAVE CONFIG.JSON

Now after editing your config.json, run official Fooocus release like normal use, then run "python foooxus.py".
Now you can enter the following address with your browser.

`http://127.0.0.1:7878`


Notes : 
-----------------------------------------------------------------------------------
Pay attention to the port number of "Fooocus" which runs on your machine.
For my system, it runs on "http://127.0.0.1:7865" (Focus not FoooXUS)

So my config.json is as follows, modify accordingly

```
"host": "127.0.0.1",        
"port": 7878,
"fooocus-directory": "C:\\Users\\user2\\Documents\\Focus",
"fooocus-address": "127.0.0.1:7865",
```

In this case, after running official focus on port 7865 (keep it running!), now you can start your FoooXus (python foooxus.py) which runs on port 7878. 

--> You are going to open browser tab with "http://127.0.0.1:7878" to reach FoooXus. Because official focus will be running on port 7865. 

I hope this helps.

**Comment by southbayjay** [comment]
Thank you!

**Comment by simartem** [comment]
If you are facing any issues on some step, put an SS or define the problem here that we can figure out. Good luck!
